crypto/fipsmodule/md5/asm/md5-x86_64.pl

```diff
@@ -52,9 +51,9 @@ sub round2_step
 	and	$x,		%r12d		/* x & z */
 	and	$y,		%r11d		/* y & (not z) */
 	mov	$k_next*4(%rsi),%r10d		/* (NEXT STEP) X[$k_next] */
-	or	%r11d,		%r12d		/* (y & (not z)) | (x & z) */
+	add	%r11d,		$dst		/* dst += (y & (not z)) */
```

Comment:
Again here, why not do this add secondly or swap lines 51 and 52?
Also for the subsequent `mov`s, are you relying on that the processor will delay them in out-of-order processing manner?

Reply:
If I understand your question correctly, yes it's similar to the above - we rely on the processor to order the independent instructions efficiently. I don't think you'd see any performance change from manual reordering.